pkgs/by-name/zo/zoho-mail-desktop/package.nix

```diff
@@ -0,0 +1,36 @@
+{
+  lib,
+  appimageTools,
+  fetchurl,
+}:
+let
+  version = "1.6.5";
+  pname = "zoho-mail-desktop";
+
+  src = fetchurl {
+    url = "https://downloads.zohocdn.com/zmail-desktop/linux/${pname}-lite-x64-v${version}.AppImage";
+    hash = "sha256-TiPf5tQnVjPpzteSxIa4RLEyYBx6ohnnxPTAG0JjnrM=";
+  };
+
+  appimageContents = appimageTools.extract { inherit pname version src; };
+in
+appimageTools.wrapType2 {
+  inherit pname version src;
+
+  extraInstallCommands = ''
```

Comment:
Wait, you're passing src to the wrap function but in the build script you're only using the `appimageContents` directly?

Reply:
Ideally src should receive the actual content then you can reference it by using the bash variable itself

Reply:
wrapType2 expects an Appimage as the src, so it can create a binary file out of it (this is automatic). I extract the appimage manually for the .desktop file and icons and install them in the extraInstallCommands. If you wouldn't mind could you review this part again with that in mind :D

Reply:
I'd suggest then to pass the extracted thing as the extracted parameter and reference internally using the bash variables.